index.js

```diff
@@ -0,0 +1,125 @@
+
+'use strict'
+const cote = require('cote')
+const u = require('@elife/utils')
+const axios = require('axios')
+const https = require('https')
+require('dotenv').config()
+
+
+var sModel = "text-davinci-003";
+var iMaxTokens = 2048;
+var sUserId = "1";
+var dTemperature = 0.5;  
+var OPENAI_API_KEY = process.env.CHATGPT_APIKEY;
+     
+const headers = {
+    'Content-Type': 'application/json',
+    "Accept": "application/json",
+    "Authorization": "Bearer " + OPENAI_API_KEY,
+
+}
+
+function main() {
+    startMicroservice()
+    registerWithCommMgr()
+}
+
+
+let msKey = 'everlife-chat-gpt'
+
+
+/*      outcome/
+ * Register ourselves as a message handler with the communication
+ * manager so we can handle requests for secret messages.
+ */
+function registerWithCommMgr() {
+    commMgrClient.send({
+        type: 'register-msg-handler',
+        mskey: msKey,
+        mstype: 'msg',
+        mshelp: [ { cmd: '/chatgpt', txt: "Get replies from OpenAIs's chatgpt server" } ],
+    }, (err) => {
+        if(err) u.showErr(err)
+    })
+}
+
+
+const commMgrClient = new cote.Requester({
+    name: 'ChatGPT Skill -> CommMgr',
+    key: 'everlife-communication-svc',
+})
+
+function sendReply(msg, req) {
+    req.type = 'reply'
+    req.msg = String(msg)
+    commMgrClient.send(req, (err) => {
+        if(err) u.showErr(err)
+    })
+}
+
+function startMicroservice() {
+
+    /*      understand/
+     * The microservice (partitioned by key to prevent
+     * conflicting with other services).
+     */
+    const svc = new cote.Responder({
+        name: 'Everlife ChatGPT Skill',
+        key: msKey,
+    })
+
+    /*      outcome/
+     * Respond to user messages asking us to code/decode things
+     */
+    svc.on('msg', (req, cb) => {
+        if(req.msg && !req.msg.startsWith('/')) {
+            cb(null, true) /* Yes I am handling this message */
+            reqChatGPTServer(req.msg.substring(''.length),cgptAnswer=>{
+                if(cgptAnswer) sendReply(cgptAnswer,req)
```

Comment:
This will cause a everlife to hang if there is a problem with the response. It needs to respond with something. So the response needs to be in standard nodejs fashion (i.e `(err, cgptAnswer)`). Then, if we get an error, we can log it and ask the user to try again or something (in a polite chatty fashion)